gson/src/test/java/com/google/gson/JsonPrimitiveTest.java

```diff
@@ -316,4 +316,16 @@ public void testDeepCopy() {
     JsonPrimitive a = new JsonPrimitive("a");
     assertThat(a).isSameInstanceAs(a.deepCopy()); // Primitives are immutable!
   }
+
+  @Test
+  public void testBigDecimalEquals() {
+    JsonPrimitive small = new JsonPrimitive(1.0);
+    JsonPrimitive large = new JsonPrimitive(2.0);
+    assertThat(small.equals(large)).isFalse();
```

Comment:
This assert probably is overkill because is already tested here:
https://github.com/google/gson/blob/3adead6ab4f240ed53322eff34210ed4a69dd490/gson/src/test/java/com/google/gson/JsonPrimitiveTest.java#L246-L250

It's here to be consistent with the code in the issue #904